service/gateway/metadata_handler.go

```diff
@@ -102,8 +105,32 @@ func (gateway *Gateway) listObjectsByBucketNameHandler(w http.ResponseWriter, r
 		return
 	}
 
+	if reqContext.maxKeys != "" {
+		maxKeys, err = strconv.ParseUint(reqContext.maxKeys, 10, 64)
+		if err != nil {
+			log.Errorw("failed to parse maxKeys", "max_keys", reqContext.maxKeys, "error", err)
+			errDescription = InvalidMaxKeys
+			return
+		}
+		// maxKeys should > 0
+		if maxKeys <= 0 {
+			log.Errorw("failed to check maxKeys", "max_keys", reqContext.maxKeys, "error", err)
+			errDescription = InvalidMaxKeys
+			return
+		}
+	}
+
+	// startAfter is an optional input, we only check its format when user input the value
+	if ok = IsHexHash(reqContext.startAfter); !ok && reqContext.startAfter != "" {
```

Comment:
What is the meaning of `startAfter`?

Reply:
Compared to the traditional limit and offset, startAfter is the specified object ID, and maxKeys is the number of objects requested by the user